packages/@react-spectrum/menu/src/MenuTrigger.tsx

```diff
@@ -56,9 +56,12 @@ function MenuTrigger(props: SpectrumMenuTriggerProps, ref: DOMRef<HTMLElement>)
   }
 
   let isMobile = useIsMobileDevice();
+  let menuTreeState = UNSTABLE_useMenuState({}, state);
+  // TODO: override menuTriggerState's .close so we clear the expandedKeysStack when clicking on the underlay or whenever
+  // Popover's .close is triggered. Alternatively, we could potentially clear it in a cleanup effect when the root menu unmounts...
+  let close = menuTreeState.closeAll;
```

Comment:
I figure aria hook users would do the same when adding submenu support to their menus.

Reply:
TODO: refactor after getting rid useMenuState.